### PR TITLE
Introduce forceKill logic

### PR DESCRIPTION
**Pull request by khous**

### CODE REVIEW

**Comment by khous** [comment]
TODO - add a section in the readme with an example and description of the behavior

**Review by mphillips-infoblox — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -75,3 +76,12 @@ Pth represents a unique string that makes sense to the strategy. For example, pt
...
 The hotload project ships with one hotload strategy: fsnotify.
+
+# Force Kill
+
+By default, the hotload driver allows for connections to be closed gracefully by the underlying driver. If your 
+application holds connections open with long running operations, this will prevent graceful switchover to new datasources.
+
+Adding `forceKill=true` to your DSN will cause the hotload driver to close the underlying connection manually when a 
+change to the connection information is detected.
+
```
Comment (mphillips-infoblox) [inline README.md:87]: Can you please add some example code showing this being used in the DSN?

**Review by mphillips-infoblox — APPROVED** [review]

File: conn.go [hunk 2]
```diff
@@ -112,7 +113,15 @@ func (c *managedConn) ResetSession(ctx context.Context) error {
 }
 
 func (c *managedConn) Close() error {
-	return c.conn.Close()
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	err := c.conn.Close()
+
+	if err == nil {
```
Comment (daniel-garcia) [inline conn.go:120]: There can be valid reasons an error is returned. For example, if the underlying driver already closed the connection.  I think here we just log the error (if any) and mark as "killed" no matter what.

File: driver.go [hunk 3]
```diff
@@ -208,6 +216,16 @@ func (cg *chanGroup) Open() (driver.Conn, error) {
 	return manConn, nil
 }
 
+func (cg *chanGroup) parseValues(vs url.Values) {
+	cg.mu.Lock()
+	defer cg.mu.Unlock()
+
+	if v, ok := vs[forceKill]; ok {
+		firstValue := v[0]
```
Comment (daniel-garcia) [inline driver.go:224]: This code works fine for the positive case. If someone mistypes the camelCase, they don't get confirmation that their "settings" were bad, however.

**Pull request merged**